src/fswAlgorithms/attGuidance/flybyPoint/_UnitTest/test_flybyPoint.py

```diff
@@ -0,0 +1,285 @@
+    unitTaskName = "unitTask"
+    unitProcessName = "unitProcess"
+
+    #  Create a sim module as an empty container
+    unitTestSim = SimulationBaseClass.SimBaseClass()
+
+    #  create the simulation process
+    testProcess = unitTestSim.CreateNewProcess(unitProcessName)
+
+    # create the dynamics task and specify the integration update time
+    simulationTimeStep = macros.sec2nano(dTsim)
+    testProcess.addTask(unitTestSim.CreateNewTask(unitTaskName, simulationTimeStep))
+
+    #   setup the FSW algorithm tasks
+
+    # setup flybyPoint guidance module
+    flybyGuid = flybyPoint.FlybyPoint()
+    flybyGuid.ModelTag = "flybyPoint"
+    flybyGuid.dtFilterData = dTfilter
+    unitTestSim.AddModelToTask(unitTaskName, flybyGuid)
+
+    inputData = messaging.NavTransMsgPayload()
+    inputData.v_BN_N = np.array(initVel)
+    filterInMsg = messaging.NavTransMsg()
+    flybyGuid.filterInMsg.subscribeTo(filterInMsg)
+
+    #
+    #   Setup data logging before the simulation is initialized
+    #
+    attRefLog = flybyGuid.attRefOutMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, attRefLog)
+
+    unitTestSim.InitializeSimulation()
+    dataPos = []
+    dataVel = []
+    for i in range(round(9*600/dTsim)):
+        dataPos.append(np.array(initPos) + np.array(initVel) * (i * dTsim))
+        dataVel.append(np.array(initVel))
+        inputData.timeTag = macros.sec2nano(i * dTsim)
+        inputData.r_BN_N = dataPos[i]
+        filterInMsg.write(inputData, unitTestSim.TotalSim.CurrentNanos)
+        unitTestSim.ConfigureStopTime(macros.sec2nano((i + 1) * dTsim) - 1)
+        unitTestSim.ExecuteSimulation()
+
+    #   retrieve the logged data
+    refAtt = attRefLog.sigma_RN
+    refRate = attRefLog.omega_RN_N
+    refAcc = attRefLog.domega_RN_N
+    timeData = attRefLog.times() * macros.NANO2MIN
+
+    ur_output = []
+    ut_output = []
+    uh_output = []
+    for i in range(len(refAtt)):
+        RN = rbk.MRP2C(refAtt[i])
+        ur_output.append(np.matmul(RN.transpose(), [1, 0, 0]))
+        ut_output.append(np.matmul(RN.transpose(), [0, 1, 0]))
+        uh_output.append(np.matmul(RN.transpose(), [0, 0, 1]))
+
+    ur = initPos / np.linalg.norm(initPos)
+    uh = np.cross(initPos, initVel) / np.linalg.norm(np.cross(initPos, initVel))
+    ut = np.cross(uh, ur)
+    f0 = np.linalg.norm(initVel) / np.linalg.norm(initPos)
+    gamma0 = np.arctan(np.dot(initVel, ur) / np.dot(initVel, ut))
+    for i in range(len(refAtt)):
+        # check a vector values
+        ur = dataPos[i] / np.linalg.norm(dataPos[i])
+        uh = np.cross(dataPos[i], dataVel[i]) / np.linalg.norm(np.cross(dataPos[i], dataVel[i]))
+        ut = np.cross(uh, ur)
+        dt = timeData[i]*60
+        den = ((f0*dt)**2 + 2*f0*np.sin(gamma0)*dt + 1)
+        omega = uh * f0 * np.cos(gamma0) / den
+        omegaDot = uh * (-2*f0*f0 * np.cos(gamma0)) * (f0*dt + np.sin(gamma0)) / den / den
+        if not unitTestSupport.isVectorEqual(ur_output[i], ur, accuracy):
```

Comment:
:small_red_triangle_down: Let's replace all these checks with numpy testing asserts. E.g. `np.testing.assert_allclose(ur_output[i], ur, rtol=0, atol=accuracy, verbose=true)`

This means you can also delete all the `testMessages` and `testFailCount` stuff.

Reply:
Done